components/script/webdriver_handlers.rs

```diff
@@ -37,15 +37,15 @@ use util::str::DOMString;
 fn find_node_by_unique_id(page: &Rc<Page>, pipeline: PipelineId, node_id: String) -> Option<Root<Node>> {
     let page = get_page(&*page, pipeline);
     let document = page.document();
-    let node = document.upcast::<Node>();
-
-    for candidate in node.traverse_preorder() {
-        if candidate.get_unique_id() == node_id {
-            return Some(candidate);
-        }
-    }
+    document.upcast::<Node>().traverse_preorder().find(|candidate| candidate.get_unique_id() == node_id)
+}
 
-    None
+fn find_element_by_element_id(page: &Rc<Page>, pipeline: PipelineId, element_id: String) -> Option<Root<Element>> {
+    let page = get_page(&*page, pipeline);
+    page.document().upcast::<Node>().traverse_preorder().find(|x| match x.downcast::<Element>() {
+        Some(elem) => elem.Id() == &*element_id,
+        None => false
+    }).map(|x| Root::from_ref(x.downcast::<Element>().unwrap()))
```

Comment:
Good point, but I think I'm wrong here. See Step 3 of the [spec](https://w3c.github.io/webdriver/webdriver-spec.html#get-element-rect). I think I'm supposed to find by UUID or `Node::unique_id`.